libraries/lib-module-manager/PluginManager.cpp

```diff
@@ -1162,43 +1163,16 @@ std::map<wxString, std::vector<wxString>> PluginManager::CheckPluginUpdates()
    // When the user enables the plugin, each provider that reported it will be asked
    // to register the plugin.
 
+   auto& moduleManager = ModuleManager::Get();
    std::map<wxString, std::vector<wxString>> newPaths;
-   for (auto &pair : mRegisteredPlugins) {
-      auto &plug = pair.second;
-      const PluginID & plugID = plug.GetID();
-      const wxString & plugPath = plug.GetPath();
-      PluginType plugType = plug.GetPluginType();
-
-      // Bypass 2.1.0 placeholders...remove this after a few releases past 2.1.0
-      if (plugType == PluginTypeNone)
-         continue;
-
-      if ( plugType == PluginTypeModule  )
-      {
-         if (!mm.IsProviderValid(plugID, plugPath))
-         {
-            plug.SetEnabled(false);
-            plug.SetValid(false);
-         }
-         else
-         {
-            if (auto provider = mm.CreateProviderInstance( plugID, plugPath ))
-            {
-               const auto paths = provider->FindModulePaths(*this);
-               for (size_t i = 0, cnt = paths.size(); i < cnt; i++)
-               {
-                  wxString path = paths[i].BeforeFirst(wxT(';'));
-                  if(!make_iterator_range(pathIndex).contains(path))
-                     newPaths[path].push_back(plugID);
-               }
-            }
-         }
-      }
-      else if (plugType != PluginTypeStub)
+   for(auto& [id, provider] : moduleManager.Providers())
+   {
+      const auto paths = provider->FindModulePaths(*this);
+      for(const auto& path : paths)
       {
-         plug.SetValid(mm.IsPluginValid(plug.GetProviderID(), plugPath, false));
-         if (!plug.IsValid())
-            plug.SetEnabled(false);
+         const auto modulePath = path.BeforeFirst(';');
```

Comment:
Not new in this commit -- but PluginManager is doing some interpretation of the paths returned by `FindModulePaths`.  So the comment in PluginProvider.h about `FindModulePaths` is not correct.  It is expected paths can be partly interpreted in a way not specific to the provider, at least when there is a semicolon.